themes/10up-theme/plopfile.js

```diff
@@ -0,0 +1,125 @@
+// https://plopjs.com/documentation/
+
+module.exports = (plop) => {
+	plop.setGenerator('UI component', {
+		description: 'UI Component with options for Partial, CSS, JS & Styleguide example',
+		prompts: [
+			{
+				type: 'input',
+				name: 'name',
+				message: 'Component name?',
+			},
+			{
+				type: 'list',
+				name: 'typePath',
+				message: 'Component type?',
+				choices: [
```

Comment:
These choices confused me at first. They're great for a demo, but I'd want to customize these per-project. I could add a note about it in the readme if that's the intention.

Reply:
@ian-pvd I figured this one would be called out :-)
I've had a lot of success with structuring components based on Atomic principles. It's especially helpful for partials that can be standalone or wired into block rendering, in which case the type is a useful contextual guide as to how components are intended to be used. I actually baked the 'no type' option back in because obviously that's a totally valid option as well and it really depends on project needs how we might choose to address this. For me I would really like to surface these options (not least because I know I will need them in future) and stick with a default of none

Reply:
Note added to readme